### ISSUE TITLE
isaac server does not work on taurus

### ISSUE THREAD

**Issue by PrometheusPi** [ComputationalRadiationPhysics/isaac#84]
I compiled isaac following @FelixTUD instructions. No errors occurred during build and the isaac server could run on the compute nodes. The server lisense to all three ports (`...58`=tcp, `...59`=web, `...60`=sim) and they seem to work: 
 - using `telnet` all port react
 - connections from all three ports are correctly opened and closed according to the server output
 - if using `telnet`on the tcp port, information is transmitted
 - after a simulation has connected, the tcp port provides information on that simulation thus, the sim port works fine too
 - the client (`interface.htm`) clearly identifies the web port as correct while others are identified as not correct - thus this appears to work too

**However**, the simulation never shows up in the client - thus observation is not possible. 

Doing the same with a server compiled on hypnos and port forwarding the sim port from PIConGPU to this server works flawlessly.

**Comment by ax3l** [comment]
> connections from all three ports are correctly opened

where did you connect from? locally? Opening a port on `localhost`/loopback is a different thing than a remotely accessible port. You might want to deploy SSH remote port forwarding to your local machine to make the port reachable by a (locally) running HTML browser-client.

```bash
# example, if running on hypnos5 on localhost
ssh -p 22 -L 2459:hypnos5:2459 huebl@uts.fz-rossendorf.de -N

# connect in your browser now to
# localhost:2459
# http://laser.plasma.ninja/isaac/interface.htm
```

**Comment by PrometheusPi** [comment]
@ax3l 
I did port forward correctly (using multiple hops if necessary) from 

 - hypnos5 -> ZIH (no double hop) 
 - my laptop -> ZIH (double hop)
 - @psychocoderHPC laptop -> ZIH 

Just to make absolutely sure that connections work, I spawned a firefox on taurus and accessed the server directly without any port forwarding. Same result (no simulation).

As written above, the connection worked and **showed up in the server output** correctly when connecting and when disconnecting. 

The tcp port is **neither used directly** by the client nor the simulation **but** allows to access simulation meta data. It showed the correct simulation data. This **simulation and server found each other**.

Just to make sure the DNS was resolved correctly I used both the Ethernet IP, the hostname and the hostname+zih-url. Always the same result. All these connections showed up in the server log. None showed the simulation. 

The **same workflow** of port forwarding worked flawlessly when the server is running on hypnos.

**Comment by ax3l** [comment]
On which headnode of Taurus are you currently running? I will connect myself and verify, since this report is far from reproducible without any reference of PIConGPU startup options, used scripts or isaac server startup commands.

**Comment by PrometheusPi** [comment]
Currently I am not running the server on taurus anymore. Please do not run test at the moment that might interfere with the running simulation, as this is a production run for next weeks conference.

**Comment by ax3l** [comment]
Please try to document all used startup commands and outputs, otherwise remote people like @FelixTUD or @theZiz have zero chance to support you.

> Please do not run test

I don't want to start anything, just want to connect-check to a running isaac server with a running PIConGPU connected to it with my own scripts, posted above.

**Comment by PrometheusPi** [comment]
The setup without the final *.cfg can be found [here](https://github.com/ComputationalRadiationPhysics/projects-electrons/tree/LPWFA_livVis). 

Both @FelixTUD and @tdd11235813 were already informed about the current status of this issue via mail and in person. (This is just for documentation.) 

I can start up the server for you - but I have no resources left to run a second PIConGPU simulation (even on a single node) to connect to the server for testing purposes.

**Comment by ax3l** [comment]
> Both @FelixTUD and @tdd11235813 were already informed about the current status of this issue via mail and in person. (This is just for documentation.)

Please write so next time, so people know that this can be ignored.

> but I have no resources left to run a second PIConGPU simulation (even on a single node)

ah too bad, a single node would be enough. which headnode?

**Comment by PrometheusPi** [comment]
No headnode is used since they limit runtime. I used a compute node. 
Currently it is `taurusi2084`. Ports are `22458` till `22460`.

**Comment by ax3l** [comment]
Connection works. You can run a sample test-source:

https://github.com/ComputationalRadiationPhysics/isaac/tree/master/example

**Comment by PrometheusPi** [comment]
I build the example but it crashes on start (I did not yet update the destination and the port) due to a floating point error 
```
Using name Example_936943
[tauruslogin6:5982 :0] Caught signal 8 (Floating point exception)
==== backtrace ====
 2 0x00000000000687bc mxm_handle_error()  /var/tmp/OFED_topdir/BUILD/mxm-3.6.3104/src/mxm/util/debug/debug.c:641
 3 0x0000000000068d0c mxm_error_signal_handler()  /var/tmp/OFED_topdir/BUILD/mxm-3.6.3104/src/mxm/util/debug/debug.c:616
 4 0x0000000000035270 killpg()  ??:0
 5 0x0000000000428d75 main()  /home/s5960712/lib/isaac/example/example.cpp:311
 6 0x0000000000021c05 __libc_start_main()  ??:0
 7 0x0000000000426e39 _start()  ??:0
===================
Gleitkomma-Ausnahme
```

Additionally it looks like it uses CUDA. Might this cause the error since I execute the test on a CPU only node?

**Comment by ax3l** [comment]
Yes, just pass `-DISAAC_CUDA=OFF -DISAAC_ALPAKA=ON -DALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLE=ON` to CMake.

**Comment by PrometheusPi** [comment]
With your extra flags I get the following error:
```
CMake Error at CMakeLists.txt:35 (ALPAKA_ADD_EXECUTABLE):
  Unknown CMake command "ALPAKA_ADD_EXECUTABLE".
```

(I tried this already via `ccmake` - got the same error.)

I am running on @FelixTUD [branch](https://github.com/FelixTUD/isaac/tree/dev-felix-bachelor).

**Comment by ax3l** [comment]
Set `-DALPAKA_ROOT=` to the location of alpaka

**Comment by PrometheusPi** [comment]
might be related to the libwebsocket issue #93 also seen on hemera - will test this again

**Comment by nicowrobel** [comment]
Solved, as taurus is dead
